src/git/cli_parser.rs

```diff
@@ -46,6 +46,84 @@ impl ParsedGitInvocation {
         v.extend(self.command_args.iter().cloned());
         v
     }
+    pub fn has_command_flag(&self, flag: &str) -> bool {
+        self.command_args.iter().any(|arg| arg == flag)
+    }
+
+    /// Returns the n-th positional argument after the command (0-indexed).
+    /// Skips all arguments that start with '-' (flags and their inline values).
+    ///
+    /// Examples:
+    /// - `git merge abc --squash` => pos_command(0) returns Some("abc")
+    /// - `git merge --squash --no-verify abc` => pos_command(0) returns Some("abc")
+    /// - `git merge abc def --squash` => pos_command(1) returns Some("def")
+    pub fn pos_command(&self, n: u8) -> Option<String> {
```

Comment:
@svarlamov trying to support parsing positional arguments

`git merge feat/abc --squash`
`git merge --squash --other-flag feat/abc --flag-after-that`

```
post_command(0) // always "feat/abc"
```

Does it look right?